archive/overlap.go

```diff
@@ -198,3 +209,199 @@ func mergeLargestChunkSpanInfos(spans []SpanInfo, order zbuf.Order) []SpanInfo {
 	}
 	return append(res, mergeSpanInfos(run, order))
 }
+
+// removeMaskedChunks performs an in-place edit of the spans input slice,
+// updating each SpanInfo to remove chunks that mask other chunks within the
+// same SpanInfo.
+// If trackMasked is true, it will return a slice of chunks that have been
+// masked and are no longer present in any SpanInfo.
+func removeMaskedChunks(spans []SpanInfo, trackMasked bool) []Chunk {
+	var maskIds []ksuid.KSUID
+	var maskedChunks map[ksuid.KSUID]Chunk
+	for i, si := range spans {
+		if len(si.Chunks) == 1 {
+			continue
+		}
+		maskIds = maskIds[:0]
+		for _, c := range si.Chunks {
+			for _, mid := range c.Masks {
+				maskIds = append(maskIds, mid)
+			}
+		}
+		if len(maskIds) == 0 {
+			continue
+		}
+		var chunks []Chunk
+		for _, c := range si.Chunks {
+			var masked bool
+			for _, mid := range maskIds {
+				if mid == c.Id {
+					masked = true
+					if trackMasked {
+						if maskedChunks == nil {
+							maskedChunks = make(map[ksuid.KSUID]Chunk)
+						}
+						maskedChunks[c.Id] = c
+					}
+					break
+				}
+			}
+			if !masked {
+				chunks = append(chunks, c)
+			}
+		}
+		// No need to sort chunks since we perform the mask removal in-order.
+		spans[i] = SpanInfo{
+			Span:   si.Span,
+			Chunks: chunks,
+		}
```

Comment:
I would move this functionality to be a receiver function of `SpanInfo`. I think it would help readability. Something like: 
```golang
func (s *SpanInfo) RemoveMasked() (removed []Chunk)
```

Reply:
👍  will do